GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace GenericsHomework;
+
+public class Node<TVal> where TVal : notnull
```

Comment:
```suggestion
public class Node<T>
```
You don't need this here, also `T` is the commonly used identifier for generics.